js/wordsearch.js

```diff
@@ -0,0 +1,315 @@
+  'use strict';
+
+ // Initialize the application
+  WordSearch.prototype.initialize = function() {
+    this.matrix     = [];                            // Letter matrix    
+    this.selectFrom = null;                         // Selection from
+    this.selected   = [];                          // Selected Letters
+    this.initmatrix(this.settings.gridSize);
+  }
+
+   /**
+   * Fill default items into the matrix
+   * @param {Number} size Grid size
+   */
+  WordSearch.prototype.initmatrix = function(size) {
+    for (var row = 0; row < size; row++) {
+      for (var col = 0; col < size; col++) {
+        var item = {
+          letter: '.',                                 // Default value
+          row: row,
+          col: col
+        }
+        if (!this.matrix[row]) this.matrix[row] = [];
+        this.matrix[row][col] = item;
+      }
+    }
+  }
+
+  // Mouse event - Mouse down
+  WordSearch.prototype.onMousedown = function(item)  { return () => this.selectFrom = item; }
+
+  // Mouse event - Mouse move
+  WordSearch.prototype.onMouseover = function(item) {
+    return () => {
+      if (this.selectFrom) {
+        this.selected = this.getItems(
+        this.selectFrom.row,
+        this.selectFrom.col,
+        item.row,
+        item.col
+        );
+        this.reset();
+        this.selected.forEach(word => {
+          var current = word,
+            row = current.row + 1,
+            col = current.col + 1,
+            wordSelected = document.querySelector('.section .ws-row:nth-child(' + row + ') .ws-col:nth-child(' + col + ')');
+          wordSelected.className += ' ws-selected';
+        })
+      }
+    }
+  }
+
+  // Mouse event - Mouse up
+  WordSearch.prototype.onMouseup = function() {
+    return () => {
+      this.selectFrom = null;
+      this.searched(this.selected);
+      this.selected = [];
+      this.reset();
+    }
+  }
+
+  // Returns a random integer between min and max
+  if (typeof Math.rangeInt != 'function') {
+    Math.rangeInt = function(minAscii, maxAscii){
+      if (!maxAscii) {
+        maxAscii = minAscii;
+        minAscii = 0;
+      }
+      return Math.floor(Math.random() * (maxAscii - minAscii + 1)) + minAscii;
+    }
+  }
+
+  // Word seach constructor
+  function WordSearch(context) {
+    this.solved = 0;                                             // Words solved.
+    this.settings = {                                           // Default settings
+      'directions'  : ['W', 'N', 'WN', 'EN'],
+      'gridSize'    : 10,
+      'words'       : ['one', 'two', 'three', 'four', 'five'],
+      'wordsList'   : [],
+      'fillLetters' : false
+    }
+    this.gridSection = context;
+    this.gridSection.classList.add('section');                      // Add `.section` to wrap element
+    // Check the words' length if it is overflow the grid
+    if (this.parseWords(this.settings.gridSize, this.settings.words.length)) {
+      var wordAddedWork = false;
+      while (!wordAddedWork) {
```

Comment:
```suggestion
      while (!this.addWords()) {
```
Can we use like this?